src/nixpacks/builder/docker.rs

```diff
@@ -15,6 +15,50 @@ use indoc::formatdoc;
 use tempdir::TempDir;
 use uuid::Uuid;
 
+const DOT_NIXPACKS_DIR: &'static &str = &".nixpacks";
+
+struct OutputDir {
+    root_path: String,
+    dockerfile_path: String,
+    environment_nix_relative_path: String,
+    environment_nix_absolute_path: String,
+}
+
+impl OutputDir {
+    pub fn new(root_path: &str) -> Result<Self> {
+        let dot_nixpacks_dir = PathBuf::from(root_path)
+            .join(PathBuf::from(DOT_NIXPACKS_DIR))
+            .display()
+            .to_string();
+
+        let dockerfile_path = PathBuf::from(&dot_nixpacks_dir)
+            .join(PathBuf::from("Dockerfile"))
+            .display()
+            .to_string();
+
+        let environment_nix_relative_path = PathBuf::from(DOT_NIXPACKS_DIR)
+            .join(PathBuf::from("environment.nix"))
+            .display()
+            .to_string();
+
+        let environment_nix_absolute_path = PathBuf::from(root_path)
+            .join(PathBuf::from(&environment_nix_relative_path))
+            .display()
+            .to_string();
```

Comment:
Instead of storing both the absolute and relative paths for this single file, how about storing JUST either the absolute or relative paths for both the dockerfile and environment file.